operator/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -333,6 +333,21 @@ func (r *RedpandaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (r
 func (r *RedpandaReconciler) reconcileResources(ctx context.Context, cluster *lifecycle.ClusterWithPools) (err error) {
 	ctx, span := trace.Start(ctx, "reconcileResources")
 	defer func() { trace.EndSpan(span, err) }()
+
+	//// The flag that disables cluster configuration synchronization is set to `true` to not
+	//// conflict with operator cluster configuration synchronization.
+	//if cluster.Spec.ClusterSpec.Statefulset == nil {
+	//	cluster.Spec.ClusterSpec.Statefulset = &redpandav1alpha2.Statefulset{}
+	//}
+	//if cluster.Spec.ClusterSpec.Statefulset.SideCars == nil {
+	//	cluster.Spec.ClusterSpec.Statefulset.SideCars = &redpandav1alpha2.SideCars{}
+	//}
+	//if cluster.Spec.ClusterSpec.Statefulset.SideCars.ConfigWatcher == nil {
+	//	cluster.Spec.ClusterSpec.Statefulset.SideCars.ConfigWatcher = &redpandav1alpha2.ConfigWatcher{
+	//		SkipClusterConfigurationSync: true,
+	//	}
+	//}
```

Comment:
I don't understand how test could pass 👎 Without setting `noSetSuperusers` flag of sidecar.

Reply:
It actually makes sense! The test wasn't passing previously because the operator was setting the wrong values. Now, we're setting the correct value from the operator _and_ the sidecar, albeit with potentially different ordering. The sidecar will only trigger once, so the window for this test to fail is actually quite low.

If you wanted to make it more strict, you could add an additional assert about the cluster configuration version. It should only increment once per change to the spec but the sidecar may cause it to increment an additional time.